app/main.py

```diff
@@ -1,2 +1,55 @@
+from typing import Any, Optional
+
+
 class Dictionary:
-    pass
+    def __init__(self, initial_capacity: int = 8) -> None:
+        self._capacity: int = initial_capacity
+        self._size: int = 0
+        self._table: list[Optional[Dictionary._Node]] = [None] * self._capacity
+
+    class _Node:
+        def __init__(self, key: Any, value: Any, hash_value: int) -> None:
+            self.key: Any = key
+            self.value: Any = value
+            self.hash: int = hash_value
```

Comment:
The `hash_value` parameter in the `_Node` constructor is unnecessary. The hash value is used internally for indexing and does not need to be stored in the node. Consider removing this parameter and the corresponding attribute.